spec/models/pro_account_spec.rb

```diff
@@ -40,32 +40,79 @@
 
   describe 'validations' do
 
+    it { is_expected.to be_valid }
+
     it 'requires a user' do
-      pro_account = FactoryBot.build(:pro_account, user: nil)
+      pro_account.user = nil
       expect(pro_account).not_to be_valid
     end
 
   end
 
-  describe 'create callbacks' do
+  describe '#update_stripe_customer' do
+
+    let(:mock_customer) { double(:customer).as_null_object }
+
+    before do
+      allow(Stripe::Customer).to receive(:new).and_return(mock_customer)
+    end
+
+    it 'stores Stripe customer ID' do
+      expect {
+        pro_account.update_stripe_customer
+      }.to change(pro_account, :stripe_customer_id).from(nil)
+    end
 
-    it 'creates Stripe customer and stores Stripe customer ID' do
-      pro_account = FactoryBot.build(:pro_account, stripe_customer_id: nil)
-      expect(Stripe::Customer).to receive(:create).and_call_original
-      pro_account.run_callbacks :create
-      expect(pro_account.stripe_customer_id).to_not be_nil
+    it 'creates Stripe customer' do
+      pro_account.update_stripe_customer
+      expect(mock_customer).to have_received(:save)
+    end
+
+    it 'sets Stripe customer email' do
```

Comment:
These specs test implementation details, not behaviour.

We should be asserting that e.g. `expect(customer.email).to eq user.email`

Reply:
Changed in bb3b1a36313b18a9426a72bd624de42e229bdc40